src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateGetHashCodeMethod.cs

```diff
@@ -166,6 +167,21 @@ internal static partial class SyntaxGeneratorExtensions
             return statements.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// In VB it's more idiomatic to write things like <c>Dim t = TryCast(obj, SomeType)</c>
+        /// instead of <c>Dim t As SomeType = TryCast(obj, SomeType)</c>, so we just elide the type
+        /// from the decl.  For C# we don't want to do this though.  We want to always include the
+        /// type and let the simplifier decide if it should be <c>var</c> or not.
+        /// </summary>
+        private static SyntaxNode SimpleLocalDeclarationStatement(
+            this SyntaxGenerator generator, INamedTypeSymbol namedTypeSymbol,
+            string name, SyntaxNode initializer)
+        {
+            return generator.RequiresLocalDeclarationType()
+                ? generator.LocalDeclarationStatement(namedTypeSymbol, name, initializer)
```

Comment:
💡 Can we file an issue to update `LocalDeclarationStatement` to apply our new attribute that adds a `using` directive when possible to simplify the explicit type reference?

Reply:
Will do.  Can you merge this in plz @sharwell ?  Thanks!

Reply:
https://github.com/dotnet/roslyn/issues/39968